File: 387_FirstUniqueCharacterInAString.c
//51.23%
int firstUniqChar(char* s) {
	if(!s)
		return -1;
    #define CHAR_NUM 256
	int counter[CHAR_NUM] = {0};
	//int indice[CHAR_NUM] = {-1};
	char *ptr = s;
	int i = 0;
	while(ptr[i] != '\0'){
		//if(!counter[ptr[i]])
		//	indice[ptr[i]] = i;
		counter[ptr[i]]++;
		i++;
	}
	i = 0;
	while(ptr[i] != '\0'){
		if(counter[ptr[i]] == 1)
			return i;
			//return indice[ptr[i]];
		i++;
	}
	return -1;			
}

File: 387_FirstUniqueCharacterInAString.py
#23%
import collections
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        counter = collections.Counter(s)
        for i in range(len(s)):
            if counter[s[i]] == 1:
                return i
        return -1
		
class Solution(object):
    def firstUniqChar(self, s):
        return min([s.find(c) for c in string.ascii_lowercase if s.count(c)==1] or [-1])
		
class Solution(object):
    def firstUniqChar(self, s):
        return min([s.find(c) for c,v in collections.Counter(s).iteritems() if v==1] or [-1])